internal/caddy/convert.go

```diff
@@ -66,7 +66,11 @@ func newConfig(namespace string, store *controller.Store) (*Config, error) {
 					MetricsServer: metricsServer(store.ConfigMap.Metrics),
 					HttpServer: {
 						AutoHTTPS: &caddyhttp.AutoHTTPSConfig{},
-						Listen:    []string{":443"},
+						// Listen to both :80 and :443 ports in order
+						// to use the same listener wrappers (PROXY protocol use it)
+						// note that order is important here because otherwise, caddy will
+						// choose the :80 port as the HTTPS port.
```

Comment:
Oof, is this true? that doesn't sound right.

I can look into it when I get a chance, but I'm pretty sure that's either not correct or it's a bug in Caddy.